xdp-loader/xdp-loader.c

```diff
@@ -101,12 +111,22 @@ int do_load(const void *cfg, __unused const char *pin_root_path)
 
 retry:
 	for (i = 0; i < num_progs; i++) {
+		DECLARE_LIBXDP_OPTS(xdp_program_opts, xdp_opts, 0);
+
 		p = progs[i];
 		if (p)
 			xdp_program__close(p);
 
-		p = xdp_program__open_file(opt->filenames.strings[i],
-					   opt->section_name, &opts);
+		if (opt->prog_name) {
+			xdp_opts.open_filename = opt->filenames.strings[i];
+			xdp_opts.prog_name = opt->prog_name;
+			xdp_opts.opts = &opts;
+
+			p = xdp_program__create(&xdp_opts);
+		} else {
+			p = xdp_program__open_file(opt->filenames.strings[i],
+						   opt->section_name, &opts);
+		}
```

Comment:
Hmm, seems a bit weird to have to do this kind of if statement; maybe we should add 'section_name' as an opts member so we can use `xdp_program__create()` for everything?